flocker/node/agents/test/test_blockdevice.py

```diff
@@ -193,14 +207,26 @@ def create_blockdevicedeployer(
 def detach_destroy_volumes(api):
     """
     Detach and destroy all volumes known to this API.
+    If we failed to detach a volume for any reason,
```

Comment:
Presumably failures might happen as exceptions? In which case the loop won't happen. So maybe add try/except that logs errors.